internal/rbd/clone.go

```diff
@@ -188,6 +189,12 @@ func (rv *rbdVolume) doSnapClone(ctx context.Context, parentVol *rbdVolume) erro
 	if err != nil {
 		return err
 	}
+	err = tempClone.unsetAllMetadata(k8s.GetSnapshotMetadataKeys())
```

Comment:
you should be resetting the parent volume metadata here isn't it? or its snapshot metadata as its done here

Reply:
YEs, good catch. Looks like it got messed up over various revisions of the patch sets. Thanks!